src/aws.sqs/amazonsqs_utils.bal

```diff
@@ -46,6 +46,12 @@ function handleResponse(http:Response|error httpResponse) returns @untainted xml
         } else {
                 return ResponseHandleFailed(RESPONSE_PAYLOAD_IS_NOT_XML_MSG);
         }
+    } else if (httpResponse is http:Payload) {
```

Comment:
Here the `http:Payload` type could have a value only if the target type defaultable param is stated in the http:Client remote function. Here the use intent to perform data binding along with the network call.

Eg : 
```
var httpResponse = self.clientEp->post(endpoint, request, targetType = xml);
```
Once the network call is done, the payload of the response is accessed as an XML payload type and returned. In this case, http:Response type will not have any value, and ideally the user should do a typing test for XML type. 

So the bottom line is, since the handleResponse method will get only the http:Response or error types, the `http:Payload` type test is unreachable. 
You can simply return or panic here.

`return Error("unreachable state : response type cannot be http payload");`